### PR TITLE
Profiles endpoints

### PR DESCRIPTION
**Pull request by jelinski**

### CODE REVIEW

File: src/main/scala/com/softwaremill/realworld/profiles/ProfileDto.scala [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.softwaremill.realworld.profiles
```
Comment (adamw) [inline src/main/scala/com/softwaremill/realworld/profiles/ProfileDto.scala:1]: I think the naming convention is that if the file doesn't correspond to a class it should be lower-case

also - do we follow the `*Dto` naming convention elsewhere in the project?

File: src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import zio.{Cause, Console, Exit, IO, ZIO, ZLayer}
+
+import scala.util.chaining.*
+
+class ProfilesEndpoints(base: BaseEndpoints, profilesService: ProfilesService) {
+
+  private val getProfile: ZServerEndpoint[Any, Any] = base.secureEndpoint.get
+    .in("api" / "profiles" / path[String]("username"))
+    .out(jsonBody[Profile])
+    .serverLogic(session => username => profilesService.getProfile(username, session.email) pipe defaultErrorMappings)
```
Comment (adamw) [inline src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala:18]: let's use dot consistently, so `.pipe`; plus, do other endpoints follow the same pattern?

File: src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    .serverLogic(session => username => profilesService.follow(username, session.email) pipe defaultErrorMappings)
+
+  private val unfollowUser: ZServerEndpoint[Any, Any] = base.secureEndpoint.delete
+    .in("api" / "profiles" / path[String]("username") / "follow")
+    .out(jsonBody[Profile])
+    .serverLogic(session => username => profilesService.unfollow(username, session.email) pipe defaultErrorMappings)
+
+  val endpoints: List[ZServerEndpoint[Any, Any]] = List(getProfile, followUser, unfollowUser)
+
+  private def defaultErrorMappings[E <: Throwable, A](io: IO[E, A]): IO[ErrorInfo, A] = io.mapError {
```
Comment (adamw) [inline src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala:32]: this looks like something that can be shared across endpoints

**Review by adamw — APPROVED** [review]

File: src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala (hunk 2 above)
Comment (FawnPug) [reply src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala:18]: It's a minor but I would also add new lines to be consistent with other files (and in my opinion this is a clearer approach)

**Review by FawnPug — APPROVED** [review]

File: src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala (hunk 2 above)
Comment (adamw) [reply src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala:18]: for that we should simply rely on scalafmt :)

File: src/main/scala/com/softwaremill/realworld/profiles/ProfileDto.scala (hunk 1 above)
Comment (jelinski) [reply src/main/scala/com/softwaremill/realworld/profiles/ProfileDto.scala:1]: Yes, I was inspired by `UserDto` file.
However, I decided to extract each class into its own dedicated file 👌

File: src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala (hunk 3 above)
Comment (jelinski) [reply src/main/scala/com/softwaremill/realworld/profiles/ProfilesEndpoints.scala:32]: Done 👍

**Pull request merged**